### DISCUSSION TITLE
Can't display correct with east asian characters in TextBox control with ui package but ok for windows title

### DISCUSSION THREAD

**Discussion by LukyGuyLucky** [vlang/v#20236] [Questions and Answers]
V cool !

When I play around the ui package,I noticed that a window title in east asian character will be displayed correctly,but when same strings in a TextBox (get input from user input) will be displayed incorrectly,somthing just like a square.Confirm the source file is in UTF8 or UTF8-BOM.

Was I missing something for proper using an Encoding function before pass the east asian  string such as  ”早上好！“ or "开心每一天~“ to the UI control?

By the way,the language reference and stdlib docs are good but could it be better?1).github is not reachable everytime in every place in the world; 2). completeness and more workable examples.I looked around but did not found any information about !  / ? in a error manipulation.Maybe my careless.

Best wishes to this totally 44 key words powerful,elegant,beautiful and simple language in the world!

**Comment by GH-TheRaven-06835** [comment]
Maybe look into this:
https://github.com/vlang/v/blob/master/examples/ttf_font/example_ttf.v

**Comment by JalonSolov** [comment]
For `!`/`?` docs, see:  https://github.com/vlang/v/blob/master/doc/docs.md#optionresult-types-and-error-handling

You are correct that the docs could be better, and we have talked about making a downloadable PDF, but that has not happened, yet, and may be a while.

@rcqls may be able to help with the UI questions.

**Comment by LukyGuyLucky** [comment]
I like very much the V philosophy,unlike some other language which wants every features included that all languages have in the world which I once followed for years and finnally got a mess.

May I ask please give a bit more focus on:

1. tutorias something structural,complete,more fundmental for very fresh beginners who want to learn a language.I came from there(and still be there:),I felt much a lot on when the new language meterials is not describled in local language,there is rare help resources,the only way when met difficulties is asking for help in the forum(but sometimes ppl not so willing to answer those silly stupid basic questions),or learn from a C/c++ book.That's a rather bad experience.So what I try to say is Learning V in V,both those new high level features but also those low level stuffs suck as ponters,memories...NOT in C,Java,C++,Pascal,Basic etc.

2. This is not a complaint,just a feedback. In windows console ,or a UI interface,or even interactive/exchange data between console and files,interop with thirdparty c libraries or OS APIs whatsoever,we always encounter a very common issue is that the new language handles rather poor(pay less attention to the subject during the development which we fully undertand) on encoding(or whatever caused.Yes we knew how to save the source file in a certain encoding format,or change windows console codepage using chcp(),even using windows API),example like I mentioned above.Some new language just replied go to MSDN,go to M$ for help,that's just an issue of OS,not an issue of the language.Yeah,I know,maybe.but at least C/C++ can figure me a way to work around and finally make it but THAT language just can't make it over a decade.This is pity ,although we like that language very much once upon a time.

Sorry for the poor expressing and best wishes these input could helps you guys a little.

Thanks and best regards,
Peter

**Comment by LukyGuyLucky** [comment]
Thank you for the help.Just read this.

Would you please show a piece of code snippet please?
<quoted>
you will need to include the fonts by reference (Windows code pages) or embed/link the font.
</quoted>
________________________________
发件人: R.D.Rush ***@***.***>
发送时间: 2023年12月25日 9:31
收件人: vlang/v ***@***.***>
抄送: LukyGuyLucky ***@***.***>; Author ***@***.***>
主题: Re: [vlang/v] Can't display correct with east asian characters in TextBox control with ui package but ok for windows title (Discussion #20236)


Title bar is controlled by Windows not the application (although you can request configurations it's DM tooling); you will need to include the fonts by reference (Windows code pages) or embed/link the font.

―

**Comment by LukyGuyLucky** [comment]
I just accidently go into here:
https://docs.vosca.dev/
V Language Documentation (aka Vlang Docs) | V Documentation<https://docs.vosca.dev/>
V Programming Language documentation
docs.vosca.dev


Wow! This is the V documentation!!Was I the only person just know there is such a wonderful place for V?

A tiny small advice： could the source snippets in the document get formatted?That will do great help for newbies like me to  understand code easier such as below:

Inline assembly
<code>
a := 100 b := 20 mut c := 0 asm amd64 { mov eax, a add eax, b mov c, eax ; =r (c) as c // output ; r (a) as a // input r (b) as b } println('a: ${a}') // 100 println('b: ${b}') // 20 println('c: ${c}') // 120

</code>

Cheers,
Peter
________________________________
发件人: JalonSolov ***@***.***>
发送时间: 2023年12月25日 10:15
收件人: vlang/v ***@***.***>
抄送: LukyGuyLucky ***@***.***>; Author ***@***.***>
主题: Re: [vlang/v] Can't display correct with east asian characters in TextBox control with ui package but ok for windows title (Discussion #20236)


For !/? docs, see: https://github.com/vlang/v/blob/master/doc/docs.md#optionresult-types-and-error-handling

You are correct that the docs could be better, and we have talked about making a downloadable PDF, but that has not happened, yet, and may be a while.

@rcqls<https://github.com/rcqls> may be able to help with the UI questions.

―

**Reply by JalonSolov** [reply]
Support for that site was abandoned, and the source deleted.

What you see is all you'll ever get... at least until the site is taken down.

**Reply by jlmess77** [reply]
Since the site was abandoned and the source is gone. You still can make a new one by using the V Documentation Generator to make it better if you want to.

https://github.com/walkingdevel/v-docs-generator

**Comment by LukyGuyLucky** [comment]
Pity! noted with thanks.
________________________________
发件人: JalonSolov ***@***.***>
发送时间: 2024年1月2日 10:30
收件人: vlang/v ***@***.***>
抄送: LukyGuyLucky ***@***.***>; Author ***@***.***>
主题: Re: [vlang/v] Can't display correct with east asian characters in TextBox control with ui package but ok for windows title (Discussion #20236)


Support for that site was abandoned, and the source deleted.

What you see is all you'll ever get... at least until the site is taken down.

―

**Comment by LukyGuyLucky** [comment]
Sorry I oversaw this very useful informatin.
I just  compiled the v-docs-generator but don't know how to use it.In the readme file it says 'npm run generate',where is npm and what is that?I am sorry if I misunderstood totally.
________________________________
发件人: Jeremy Messenger ***@***.***>
发送时间: 2024年1月4日 6:30
收件人: vlang/v ***@***.***>
抄送: LukyGuyLucky ***@***.***>; Author ***@***.***>
主题: Re: [vlang/v] Can't display correct with east asian characters in TextBox control with ui package but ok for windows title (Discussion #20236)


Since the site was abandoned and the source is gone. You still can make a new one by using the V Documentation Generator to make it better if you want to.

https://github.com/walkingdevel/v-docs-generator

―

**Reply by jlmess77** [reply]
Install Node.js, it has npm included.

npm: https://docs.npmjs.com/
Node.js: https://nodejs.org/

**Comment by rcqls** [comment]
> V cool !
> 
> When I play around the ui package,I noticed that a window title in east asian character will be displayed correctly,but when same strings in a TextBox (get input from user input) will be displayed incorrectly,somthing just like a square.Confirm the source file is in UTF8 or UTF8-BOM.
> 
> Was I missing something for proper using an Encoding function before pass the east asian string such as ”早上好！“ or "开心每一天~“ to the UI control?
> 
> By the way,the language reference and stdlib docs are good but could it be better?1).github is not reachable everytime in every place in the world; 2). completeness and more workable examples.I looked around but did not found any information about ! / ? in a error manipulation.Maybe my careless.
> 
> Best wishes to this totally 44 key words powerful,elegant,beautiful and simple language in the world!

Hi, do you have font in ttf format? Textbox and any other text in v ui is managed by sfons.
